vibing-ai/marvel-ai-backend
# new refactor for the main CI/CD pipeline

## Description

This PR refactors the CI/CD pipeline by updating the `main.yml` file for GitHub Actions to resolve the authentication issue with the API key in the production environment. The motivation behind this change is to ensure smooth deployment workflows and eliminate authentication-related failures during production builds.

## Related Issue

No issue is related with this PR.

## Type of Change


## Proposed Solution

The solution involved:
- Modifying the `main.yml` file in GitHub Actions to correctly handle environment secrets, specifically the API key.
- Ensuring secure retrieval and usage of the API key in the deployment workflow.
- Testing the pipeline changes to confirm successful authentication in production.

## How to Test

1. Trigger the GitHub Actions workflow manually or by pushing changes to the repository.
2. Confirm the successful completion of the deployment step without authentication errors.
3. Validate the production environment to ensure the deployed application is functioning correctly.

## Unit Tests

No unit tests were directly added or modified since this is a CI/CD pipeline change. Testing was performed through workflow execution.

## Documentation Updates


If yes, describe what documentation updates are needed and link to the relevant documentation.

## Checklist


## Additional Information

This update resolves a critical issue affecting production deployments. The refactor ensures a smoother, more secure CI/CD process with properly handled secrets for authentication.
